collections/_article/progress-report-web-export-in-4-3.md

```diff
@@ -0,0 +1,298 @@
+or for some other web game platforms which don't allow configuring headers for `SharedArrayBuffer` support.
+
+### Backporting the single-threaded feature from Godot 3.x
+At the end of the development of Godot 4.2, [the Foundation](https://godot.foundation/) tasked me to find a way to backport
+the ability to build Godot without using threads in order to export Godot games without the pesky
+<abbr title="Cross-Origin-Opener-Policy">COOP</abbr>/<abbr title="Cross-Origin-Embedder-Policy">COEP</abbr> headers.
+
+This was not easy because 4.0 was made with threads in mind. It maybe explains why we were so hesitant to go ahead with
+this change. Fortunately, with the help of our thread guru
+[Pedro J. Estébanez (@RandomShaper)](https://github.com/RandomShaper), I was able to tame the beast and put back single-
+threaded builds.
+
+[My <abbr title="Pull Request">PR</abbr>](https://github.com/godotengine/godot/pull/85939) was finally merged at the
+beginning of the Godot 4.3 development period.
+
+![Web export thread support option](/storage/blog/progress-report-web-export-in-4-3/thread-support-export-option.webp)
+
+It even had some unexpected benefits. Apple devices (macOS and iOS) were long known to have issues when playing Godot
+Web exports. Well, when you do export your game single-threaded, these issues fortunately disappear.
+
+However, it came with a downside too: it introduced a new, virtually unsolvable bug:
+
+The single-threaded build introduced garbles in the audio of games, making them unplayable on low-end machines such as
+laptops or phones.
+
+When I mean unplayable on low-end machines, I mean it.
+[Try it yourself.](https://adamscott.github.io/truck-town-demo-main-thread/){:target="_blank"}
+(Consider yourself lucky if it doesn't glitch out.)
+
+Yes. Web single-threaded audio stream playback with Godot 4.2 is that bad. It is marginally better on Godot 3.x, but the
+problem exist there as well.
+
+## Sample playback to the rescue!
+[Here comes my <abbr title="Pull Request">PR</abbr>](https://github.com/godotengine/godot/pull/91382) (not yet merged at
+the time of writing) for the rescue. [Here's the same demo without sound issues.](https://adamscott.github.io/truck-town-main-thread-samples/){:target="_blank"}
+
+### What are samples?
+Using samples for games is not a novel idea. In fact, for a long time, it was the _de facto_ way to play sound and music
+on game consoles. As they didn't have the processing power to do the mixing on the CPU, you could send audio clips
+(hence the term "sample") to a special chip and tell it how and when to play them.
+
+Since the audio chip was operating on it's own, the sound would never lag nor glitch out if the game is lagging.
+
+### (Re)introducing samples to Godot
+When open sourced, the Godot engine offered two ways to emit sound and music: streams and samples. Sample support was used for
+the Web exports and for platforms such as the <abbr title="PlayStation Portable">PSP</abbr>.
```

Comment:
I'm not sure that's the whole story. AFAIK back in 2.1 you used Samples for sfx (wav) and Streams for music (ogg), on all platforms.

Reply:
Oh, interesting.